test/net/sourceforge/kolmafia/request/RequestTestBase.java

```diff
@@ -32,4 +82,105 @@ protected void expectSuccess(GenericRequest spy, String responseText) {
         .when(spy)
         .externalExecute();
   }
+
+  /** Set {@link #factory} to generate the given response if the path contains {@code text}. */
+  protected ArgumentMatcher<URL> respondIfContains(final String text, final String response) {
+    return respondIfContains(Collections.singletonList(text), response);
+  }
+
+  /**
+   * Set {@link #factory} to generate the given response if the path contains every element of
+   * {@code texts}.
+   */
+  protected ArgumentMatcher<URL> respondIfContains(
+      final List<String> texts, final String response) {
+    return respondIfMatches(
+        path -> {
+          for (final String text : texts) {
+            if (!path.contains(text)) {
+              return false;
+            }
+          }
+          return true;
+        },
+        response);
+  }
+
+  /** Set {@link #factory} to generate the given response if the path matches {@code pattern}. */
+  protected ArgumentMatcher<URL> respondIfMatches(final Pattern pattern, final String response) {
+    return respondIfMatches(Collections.singletonList(pattern), response);
+  }
+
+  /**
+   * Set {@link #factory} to generate the given response if the path matches every element of {@code
+   * patterns}.
+   */
+  protected ArgumentMatcher<URL> respondIfMatches(
```

Comment:
I'm not sure I understand the use case for all of these overloads.

You want to check that the path matches multiple regex patterns? and/or multiple strings? Are you trying to match multiple query parameters that may show up in an uncertain order?

Reply:
The plan is to expect that people will want to match various parameters in URLs.

We don't want to force an "order" to those parameters.
For example,
```java
{"inventory.php"},
{"which=2"}
```
can match "inventory.php?which=2&pwd=\<pwd\>"
as well as "inventory.php?pwd=\<pwd\>&which=2"

Reply:
I would prefer matchers for query parameters in that case.

For instance, something like:

    argThat(allOf(hasPath("inventory.php"), hasParam("which", 2)))

Although, even that is incorrect if you've got POST parameters that aren't part of the query string but are instead in the POST body.

Reply:
In particular, checking for the substring "inventory.php" will incorrectly match a page "subinventory.php", and checking for "which=2" will incorrectly match "which=20" and "itemwhich=2".

Reply:
Mh... That seems possible + an interesting challenge.

This might take the rest of the day, at least. Is it possible to mark these as "changes requested" so that I can notify when done?